src/nvim/menu.c

```diff
@@ -1308,6 +1308,9 @@ void ex_emenu(exarg_T *eap)
      * for exclusive mode */
     if (*p_sel == 'e' && gchar_cursor() != NUL)
       ++curwin->w_cursor.col;
+  } else if (State & CMDLINE) {
```

Comment:
What happens if `eap->addr_count` is non-zero while `(State & CMDLINE)`? Is that possible? Does line 1265 also need to be changed to `else if (!(State & CMDLINE) && eap->addr_count)` or is it ok for `mode` to be `"Visual"` in that case?